### ISSUE TITLE
Add new attr for Homebrew Cask name

### ISSUE THREAD

**Issue by Brantone** [sous-chefs/java#544]
Because Oracle likes to hide things behind login wall, naturally push is to go towards `adoptopenjdk` , this should extend to macOS + Homebrew.

At present Homebrew is hardcode to `java` in https://github.com/sous-chefs/java/blob/master/recipes/homebrew.rb#L6
Propose a new attribute to allow for passing cask name like `adoptopenjdk`.
Will do up a PR, but looking for idea/input.

**Comment by damacus** [comment]
This is now in the list of TODO, if you are able to add MacOS testing support for this cookbook, that would be great. As we can then test AdoptOpenJDK on MacOS after #589 has been merged and release. 


Thanks!

**Comment by Brantone** [comment]
Will see what I can do, might have to dust off code from last year 👍

**Comment by damacus** [comment]
Ah looks like we just need to tap something slightly different. 

https://github.com/AdoptOpenJDK/homebrew-openjdk

It might be worth holding off until I've got v8 out of the door tomorrow/Monday, as I'm just taking a scortched earth approach with this one and moving everything to custom resources.

**Comment by damacus** [comment]
I think we now want to do the same thing as #564 and allow those resources to install brews for Mac too

**Comment by damacus** [comment]
I'm going to open a new issue for that 👍
